allauth/socialaccount/providers/oauth2/provider.py

```diff
@@ -13,6 +15,13 @@ def get_login_url(self, request, **kwargs):
             url = url + "?" + urlencode(kwargs)
         return url
 
+    def get_pkce_params(self):
+        settings = self.get_settings()
+        if settings.get("OAUTH_PKCE_ENABLED", False):
```

Comment:
What do you think about adding a `pkce_enabled = False` property to the `OAuth2Provider` class and defaulting to it instead? That way, when setting the provider you'll only need to change that property instead of [having to change the `get_setting()` method](https://github.com/startupgrind/Platform/pull/11513/files#diff-f3dd0fd882038e109cd021d71752017c2207e5c628d6382d2baaeeed8dc42e38R133-R140) just to set that property.

Reply:
Something like this?
```
class OAuth2Provider(Provider):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        settings = self.get_settings()
        self.oauth_pkce_enabled = settings.get("OAUTH_PKCE_ENABLED", False)

    def get_pkce_params(self):
        if self.oauth_pkce_enabled:
            pkce_code_params = generate_code_challenge()
            return pkce_code_params
        return {}
```

Reply:
Yup, something like that. Or, simply:
```python
class OAuth2Provider(Provider):
    oauth_pkce_enabled = False

    def get_pkce_params(self):
        if settings.get("OAUTH_PKCE_ENABLED", self.oauth_pkce_enabled):
            pkce_code_params = generate_code_challenge()
            return pkce_code_params
        return {}
```

Reply:
Ah, I see what you mean!